store/dbstore.go

```diff
@@ -61,16 +62,50 @@ func NewDBStore(dirpath string) *DBStore {
 	}
 
 	var noSyncFlag uint = 0
+	currentDbPath := path.Join(dirpath, DBStoreName)
+	log.Debugf("currentDbPath:%s", currentDbPath)
 	if LmdbNoSync {
 		noSyncFlag = lmdb.NoSync
 	}
 	if err := env.Open(
-		path.Join(dirpath, DBStoreName),
+		currentDbPath,
 		lmdb.NoSubdir|noSyncFlag,
 		0600,
 	); err != nil {
 		log.Errorf("Failed to open DB environment: %v", err)
-		return nil
+		if env != nil {
+			env.Close()
+		}
+		if _, e := os.Stat(currentDbPath); e == nil {
+			brokenDbPath := currentDbPath + dbFileBrokenPrefix
+			log.Debugf("brokenDbPath:%s", brokenDbPath)
+			log.Infof("Moving %s to %s", currentDbPath, brokenDbPath)
+			_ = os.RemoveAll(brokenDbPath)
+			renameErr := os.Rename(currentDbPath, brokenDbPath)
+			if renameErr != nil {
+				log.Errorf("Failed to move %s to %s: %v", currentDbPath, brokenDbPath, err)
+				return nil
+			}
+			newEnv, err := lmdb.NewEnv()
+			if err != nil {
+				log.Errorf("Failed to create DB environment: %v", err)
```

Comment:
```suggestion
				log.Errorf("Failed to create the DB environment: %v", err)
```

Reply:
fixed, but we do not refer to it as "the" anywhere in the logs.